### PR TITLE
Formatting with prettier

### PR DESCRIPTION
**Pull request by dead10ck**
* Adds prettier to project dependencies
* Formats existing code
* Updates dependencies
* Updates generated parser
* Adds CI for running tests and ensuring formatting

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -970,381 +948,476 @@ module.exports = grammar({
...
-        "mib", "miB", "mIB", "mIb", "Mib", "MIb", "MIB",
-        "gib", "giB", "gIB", "gIb", "Gib", "GIb", "GIB",
-        "tib", "tiB", "tIB", "tIb", "Tib", "TIb", "TIB",
-        "pib", "piB", "pIB", "pIb", "Pib", "PIb", "PIB",
-        "eib", "eiB", "eIB", "eIb", "Eib", "EIb", "EIB",
-        "zib", "ziB", "zIB", "zIb", "Zib", "ZIb", "ZIB",
-    ])
+  return choice(
+    ...[
+      "b",
```
[the remark this answers is not in the document]
Comment (dead10ck) [reply grammar.js:1340]: It looks like you can selectively ignore parts of the code. I'll add that to this.

[the remark this answers is not in the document]
Comment (fdncred) [reply grammar.js]: much better, thanks for investigating this.

**Comment by fdncred** [comment]
I'm wondering if prettier is causing the latest PR to puke because of line endings or something? Any ideas?

**Comment by siph** [comment]
@fdncred It looks like an issue with `distutils` and python `3.12` which the mac build is using: https://github.com/nodejs/node-gyp/issues/2869

The linux build passes with an older version of python.

**Pull request merged**